### PR TITLE
[DOP-21445] Add XML integration tests

### PR DESCRIPTION
**Pull request by maxim-lixakov**
## Change Summary

Added integration tests for `XML` file format

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [x] Tests pass on CI and coverage does not decrease
* [ ] Documentation reflects the changes where applicable
* [ ] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/syncmaster/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: syncmaster/worker/spark.py [hunk 1]
```diff
@@ -37,7 +37,12 @@ def get_worker_spark_session(
 
 def get_packages(db_type: str) -> list[str]:
     from onetl.connection import MSSQL, Clickhouse, MySQL, Oracle, Postgres, SparkS3
-    from onetl.file.format import Excel
+    from onetl.file.format import XML, Excel
+
+    # see supported versions from https://mvnrepository.com/artifact/com.crealytics/spark-excel
+    file_formats_spark_packages: list[str] = XML.get_packages(spark_version="3.5.1") + Excel.get_packages(
```
Comment (dolfinus) [inline syncmaster/worker/spark.py:43]: I don't get why spark_version is hardcoded for XML

Comment (dolfinus) [reply syncmaster/worker/spark.py:43]: For Excel you can update URL to https://github.com/nightscape/spark-excel/issues/902

File: docker-compose.test.yml [hunk 2]
```diff
@@ -234,7 +234,7 @@ services:
       HIVE_METASTORE_DB_DRIVER: org.postgresql.Driver
       HIVE_METASTORE_DB_USER: test_hive
       HIVE_METASTORE_DB_PASSWORD: test_hive
-    profiles: [hive, hdfs, all]
+    profiles: [hive, hdfs, s3, all]
```
Comment (dolfinus) [inline docker-compose.test.yml:237]: Please add a note why this container is needed for S3 tests

File: syncmaster/worker/spark.py (hunk 1 above)
Comment (dolfinus) [reply syncmaster/worker/spark.py]: I don't mean make some changes to `spark_version="..."` format. I mean just add a comment why Spark version is hardcoded here, linking this issue.

**Review by IlyasDevelopment — APPROVED** [review]

**Comment by IlyasDevelopment** [comment]
Parhaps this can be extended: https://github.com/MobileTeleSystems/syncmaster/pull/149/files#:~:text=if%20self.transfer_dto.file_format.__class__.__name__%20%3D%3D%20%22Excel%22%3A

File: syncmaster/worker/spark.py [hunk 3]
```diff
@@ -55,10 +61,11 @@ def get_packages(db_type: str) -> list[str]:
 
         spark_version = pyspark.__version__
         # see supported versions from https://mvnrepository.com/artifact/com.crealytics/spark-excel
```
Comment (dolfinus) [inline syncmaster/worker/spark.py:63]:
```suggestion
```

**Pull request merged**